.github/workflows/release.yaml

```diff
@@ -0,0 +1,56 @@
+# Makes an automated release
+#
+# To trigger this workflow, ensure the CHANGELOG is up to date, then push the commit 
+# to be released to a `release/*` branch.
+#
+# e.g.
+#     git checkout -b release/v1.0.0
+#     git push
+#
+name: Release
+
+on:
+  push:
+    branches: [release/*]
+
+jobs:
+  create-release:
+    name: Create GitHub Release and tag
+    runs-on: ubuntu-latest
+    environment: Release
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cucumber-actions/create-release@v1.1.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+  
+  publish-rubygem:
+    name: Publish Ruby Gem
+    needs: create-release
+    runs-on: ubuntu-latest
+    environment: Release
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.0.2
+          bundler-cache: true
+      - uses: cucumber-actions/publish-rubygem@v1.0.0
```

Comment:
Can we use https://github.com/marketplace/actions/publish-to-rubygems instead? Less stuff for us to maintain?

Reply:
@aurelien-reeves [pointed out](https://github.com/cucumber/build/issues/42#issuecomment-888122381) that including 3rd party actions in our release chain is a potential supply-chain attack vector. For simple, stable, stuff like this I think it makes sense to roll our own.

I reviewed that particular one before deciding to write our own, and it's lame. It doesn't actually do the gem publish, it just [invokes a rake task](https://github.com/cadwallion/publish-rubygems-action/blob/master/entrypoint.sh#L20). It also writes the token to disk, and needs a github token for a reason I don't understand.

So I like ours better.